### DISCUSSION TITLE
Discussion: TS26.512 - How is consumption reporting triggered?

### DISCUSSION THREAD

**Discussion by shilinding** [5G-MAG/rt-5gms-media-session-handler#36] [General]
# Problem description

The conditions for submitting a consumption report to the 5GMSd AF defined in TS26.512-h40 clause 4.7.4 are:

> If the consumption reporting procedure is activated, the Media Session Handler shall submit a consumption report to the 5GMSd AF when any of the following conditions occur:
> 1. Start of consumption of a downlink streaming session;
> 2. Stop of consumption of a downlink streaming session;
> 3. Upon determining the need to report ongoing 5GMS consumption at periodic intervals determined by the `clientConsumptionReportingConfiguration.reportingInterval` property.
> 4. Upon determining a location change, if the `clientConsumptionReportingConfiguration.locationReporting` property is set to True.
> 5. Upon determining an access network change (e.g. unicast to eMBMS, or vice versa), if the `clientConsumptionReportingConfiguration.accessReporting` property is set to True.

Questions:
- About 1 and 2, in Exoplayer, when `PlaybackState` changed to `Player.STATE_BUFFERING` (**or it should be `Player.STATE_READY`?**) it indicates the starting of the session, and `PlaybackState` Changed to `Player.STATE_ENDED` indicates the stopping of the session. Correct?
- About 4 and 5, how can we know the changing of the configuration, do we request `ServiceAccessInformation` from AF and get configuration from it and check if the two configuration setting to true all the time? I don't think so.

**Comment by rjb1000** [comment]
> About 4 and 5, how can we know the changing of the configuration, do we request `ServiceAccessInformation` from AF and get configuration from it and check if the two configuration setting to true all the time? I don't think so.

The Service Access Information retrieved from the 5GMS AF at reference point M5 will be accompanied by a time-to-live value in the `Cache-Control: max-age` HTTP response header. For a long-lived media streaming session (i.e., one that goes on longer than the Service Access Information's time-to-live period), a sensible implementation of the Media Session Handler would re-request the Service Access Information from the 5GMS AF and start or stop consumption reporting according to the up-to-date configuration.

A high quality implementation of the Media Session Handler would re-request the Service Access Information using a conditional HTTP `GET` request, i.e. one that uses the `If-Modified` request header with an `ETag` value previously returned by the 5GMS AF or an `If-Modified-Since` request header with the `Last-Modified` value of the previous response. This then allows the Media Session Handler to receive a 304 _Not Modified_ response if there is no change to the Service Access Information, and just reset the time-to-live value per the new `Cache-Control: max-age` HTTP response header.

The reference implementation of the 5GMS AF should provide both `ETag` and `Last-Modified` response headers at reference point M5. Please raise a bug issue against the AF if you find different.

**Comment by rjb1000** [comment]
Is this issue resolved to your satisfaction, @shilinding?
